giskard/core/model_validation.py

```diff
@@ -12,15 +12,15 @@
 from giskard.datasets.base import Dataset
 from giskard.ml_worker.testing.registry.slicing_function import SlicingFunction
 from giskard.models.base import BaseModel, WrapperModel
+from .dataset_validation import validate_optional_target
 from ..utils import fullname
 from ..utils.analytics_collector import analytics, get_dataset_properties, get_model_properties
-from .dataset_validation import validate_optional_target
 
 
 @configured_validate_arguments
 def validate_model(model: BaseModel, validate_ds: Optional[Dataset] = None, print_validation_message: bool = True):
     try:
-        if model.meta.model_type != SupportedModelTypes.TEXT_GENERATION and validate_ds is not None:
+        if validate_ds is not None:
             validate_optional_target(validate_ds)
```

Comment:
what I mean is that we should restore this to dataset validation, and put target=None in the constructor of the BaseModel if model_type=="text_generation".

Let me know if it's unclear or if you're not convinced, we can jump on a quick call.